llama_index/llama_dataset/evaluator_evaluation.py

```diff
@@ -0,0 +1,435 @@
+    """Evaluation example prediction class.
+
+    Args:
+        feedback (Optional[str]): The evaluator's feedback.
+        score (Optional[float]): The evaluator's score.
+    """
+
+    feedback: str = Field(
+        default_factory=str,
+        description="The generated (predicted) response that can be compared to a reference (ground-truth) answer.",
+    )
+    score: Optional[float] = Field(
+        default=None,
+        description="The generated (predicted) response that can be compared to a reference (ground-truth) answer.",
+    )
+    invalid_prediction: bool = Field(
+        default=False, description="Whether or not the prediction is a valid one."
+    )
+    invalid_reason: Optional[str] = Field(
+        default=None, description="Reason as to why prediction is invalid."
+    )
+
+    @property
+    def class_name(self) -> str:
+        """Data example class name."""
+        return "EvaluatorExamplePrediction"
+
+
+class LabelledEvaluatorDataExample(BaseLlamaDataExample):
+    """Evaluation example class.
+
+    This data class contains the ingredients to perform a new "prediction" i.e.,
+    evaluation. Here an evaluator is meant to evaluate a response against an
+    associated query as well as optionally contexts.
+
+    Args:
+        query (str): The user query
+        query_by (CreatedBy): Query generated by human or ai (model-name)
+        contexts (Optional[List[str]]): The contexts used for response
+        answer (str): Answer to the query that is to be evaluated.
+        answer_by: The reference answer generated by human or ai (model-name).
+        ground_truth_answer (Optional[str]):
+        ground_truth_answer_by (Optional[CreatedBy]):
+        reference_feedback (str): The reference feedback evaluation.
+        reference_score (float): The reference score evaluation.
+        reference_evaluation_by (CreatedBy): Evaluation generated by human or ai (model-name)
+    """
+
+    query: str = Field(
+        default_factory=str, description="The user query for the example."
+    )
+    query_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the query."
+    )
+    contexts: Optional[List[str]] = Field(
+        default_factory=None,
+        description="The contexts used to generate the answer.",
+    )
+    answer: str = Field(
+        default_factory=str,
+        description="The provided answer to the example that is to be evaluated.",
+    )
+    answer_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the answer."
+    )
+    ground_truth_answer: Optional[str] = Field(
+        default=None,
+        description="The ground truth answer to the example that is used to evaluate the provided `answer`.",
+    )
+    ground_truth_answer_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the ground-truth answer."
+    )
+    reference_feedback: Optional[str] = Field(
```

Comment:
ah there the field I expected to see in the docs 👀

Reply:
Yeah hehe sorry about that!